neofs/neofs_contract.go

```diff
@@ -212,24 +220,26 @@ func InnerRingCandidateAdd(key interop.PublicKey) {
 	}
 
 	c := common.IRNode{PublicKey: key}
-	candidates := getNodes(ctx, candidatesKey)
-
-	list, ok := addNode(candidates, c)
-	if !ok {
+	if hasNode(ctx, []byte(candidatesKey), c) {
 		panic("irCandidateAdd: candidate already in the list")
 	}
 
 	from := contract.CreateStandardAccount(key)
 	to := runtime.GetExecutingScriptHash()
-	fee := getConfig(ctx, candidateFeeConfigKey).(int)
+	fee := getConfig(ctx, CandidateFeeConfigKey).(int)
 
 	transferred := gas.Transfer(from, to, fee, []byte(ignoreDepositNotification))
 	if !transferred {
 		panic("irCandidateAdd: failed to transfer funds, aborting")
 	}
 
+	stKey := append([]byte(candidatesKey), key...)
+	if storage.Get(ctx, stKey) != nil {
+		panic("irCandidateAdd: candidate already in the list")
+	}
```

Comment:
Considering comment about `hasNode`, I am a bit confused. Do we need to check it there if `hasNode` should do it instead? Maybe we do not need `hasNode` at all with this check?

Reply:
Removed.